totalRP3/modules/dashboard/DashboardTabFrame.lua

```diff
@@ -0,0 +1,330 @@
+function TRP3_DashboardHTMLContentMixin:GetText()
+	return self.text;
+end
+
+--- Called when a hyperlink in the frame is clicked.
+function TRP3_DashboardHTMLContentMixin:OnHyperlinkClick(url, text, button)
+	-- If there's a custom URL handler installed that matches this, dispatch.
+	local handler = self.urlHandlers[url];
+	if handler then
+		return handler(url, text, button);
+	end
+
+	-- Otherwise try some builtins.
+	if url:sub(1, 7) == "twitter" then
+		-- Twitter link.
+		return self:OnHyperlinkClickTwitter(url, text, button);
+	end
+
+	-- Fallback: Allow the user to just copy the URL.
+	Popups:OpenURL(url, loc.UI_LINK_SAFE);
+end
+
+function TRP3_DashboardHTMLContentMixin:OnHyperlinkClickTwitter(url, text, button)
+	-- Left click opens the Social UI, right-click gives you a profile link.
+	if Social_ToggleShow and button == "LeftButton" then
+		Social_ToggleShow(url:gsub("twitter", "|cff61AAEE@") .. "|r ");
+	else
+		url = url:gsub("twitter", "http://twitter.com/");
+		Popups:OpenURL(url, "|cff55aceeTwitter profile|r\n");
+	end
+end
+
+--- Called when any hyperlinks in the content are moused-over.
+function TRP3_DashboardHTMLContentMixin:OnHyperlinkEnter(url, text)
+	local tooltip = self.tooltip;
+
+	tooltip:Hide();
+	tooltip:SetOwner(self, "ANCHOR_CURSOR");
+
+	if Social_ToggleShow and url:sub(1, 7) == "twitter" then
+		-- Display the Twitter handle of the user and a two-line instruction
+		-- with left and right click actions.
+		tooltip:AddLine(url:gsub("twitter", "|cff61AAEE@"), 1, 1, 1, true);
+		tooltip:AddLine(strformat(
+			"|cffffff00%s:|r %s|n|cffffff00%s:|r %s",
+			loc("CM_CLICK"), loc("CM_TWEET"),
+			loc("CM_R_CLICK"), loc("CM_TWEET_PROFILE")
+		), 1, 1, 1, true);
+	else
+		-- Display the text of the hyperlink and a single left-click
+		-- line that clicking this means opening a popup.
+		tooltip:AddLine(text, 1, 1, 1, true);
+		tooltip:AddLine(strformat(
+			"|cffffff00%s:|r %s",
+			loc("CM_CLICK"), loc("CM_OPEN")
+		), 1, 1, 1, true);
+	end
+	tooltip:Show();
+end
+
+--- Called when any hyperlinks in the content are no longer moused-over.
+function TRP3_DashboardHTMLContentMixin:OnHyperlinkLeave()
+	self.tooltip:Hide();
+end
+
+--- Registers a handler for the given URL. When clicked, this handler will
+--  be called. Only a single handler may be present for any URL at a given time.
+function TRP3_DashboardHTMLContentMixin:RegisterHyperlink(url, handler)
+	self.urlHandlers[url] = handler;
+end
+
+--- Unregisters any active handler for the given URL.
+function TRP3_DashboardHTMLContentMixin:UnregisterHyperlink(url)
+	self.urlHandlers[url] = nil;
+end
+
+--- Unregisters all registered URL handler functions.
+function TRP3_DashboardHTMLContentMixin:UnregisterAllHyperlinks()
+	twipe(self.urlHandlers);
+end
+
+--- Replacement handler for SetText() which records the text for retrieval
+--  via GetText().
+function TRP3_DashboardHTMLContentMixin:setTextHooked(text)
```

Comment:
Wouldn't hooking it in this way cause the `SaveText` to be invoked for every `SimpleHTML:SetText()` call, including those outside our widget, since the SetText method is itself stored on the shared metatable for a frame?

Reply:
No, as you hook the `SetText` of `self`, first argument. It will only hook the method for that specific instance. 
I am already using this in the `ChatLinkActionButton` to update the width of the button when we change its text, and I can confirm it doesn't hook all `SetText` of all Buttons ^^
You can see that usage here https://github.com/Ellypse/Total-RP-3/blob/master/totalRP3/modules/ChatLinks/ChatLinkActionButton.lua#L73

Reply:
Perfect then, cleans that bit up nicely 😄 .